src/jimgw/samplers/blackjax/smc.py

```
@@ -0,0 +1,855 @@
+
+        # Resume from checkpoint if one exists.
+        if (
+            ckpt_path is not None
+            and config.checkpoint_interval > 0
+            and ckpt_path.exists()
+        ):
+            try:
+                with open(ckpt_path, "rb") as _f:
+                    _ckpt = pickle.load(
+                        _f
+                    )  # Only load trusted checkpoints — pickle executes arbitrary code.
```

Ruff (0.15.15)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)